tags/permissions.py

```diff
@@ -0,0 +1,126 @@
+from django.core.exceptions import SuspiciousOperation
+from rest_framework import permissions
+
+from associations.models import (
+    Association,
+    Role,
+    Product,
+    Page,
+    Loanable,
+    Loan,
+    Folder,
+    File,
+    Event,
+    Choice,
+    User,
+)
+from associations.permissions.base_permissions import _get_role_for_user
+from forum.models import Theme, MessageForum, Topic
+from tags.models import Namespace, Tag
+
+
+def get_parent_object(obj):
+    """ returns the 'parent' object ie the one on which the scope is applied """
+
+    mappers = {
+        # Associations
+        Association: lambda x: x,
+        Choice: lambda x: x.election.association,
+        Event: lambda x: x.association,
+        File: lambda x: x.association,
+        Folder: lambda x: x.association,
+        Loan: lambda x: x.loanable.library.association,
+        Loanable: lambda x: x.library.association,
+        Page: lambda x: x.association,
+        Product: lambda x: x.marketplace.association,
+        Role: lambda x: x.association,
+        # Forum
+        Theme: lambda x: x,
+        Topic: lambda x: x.theme,
+        MessageForum: lambda x: x.topic.theme,
+    }
+
+    mapper = mappers.get(obj.__class__)
+    if mapper:
+        return mapper(obj)
+    else:
+        raise Exception("No context associated to {}".format(obj))
+
+
+def can_manage_tags_for(user, instance):
+    parent = get_parent_object(instance)
+
+    if isinstance(parent, Association):
+        role = _get_role_for_user(user, parent)
+        return role and role.is_admin
+    elif isinstance(parent, Theme):
+        return user.is_admin
+    else:
+        raise NotImplementedError("Model {} is not supported".format(instance))
+
+
+def user_can_link_tag_to(user: User, tag: Tag, instance):
+    parent = get_parent_object(instance)
+    if (
+        not isinstance(parent, Namespace.SCOPES[tag.namespace.scope])
+        or parent.id != tag.namespace.scoped_to
+    ):
+        return False
+
+    if isinstance(parent, Association):
+        role = _get_role_for_user(user, parent)
```

Comment:
Sorry, forgot to tell you: the `User` model now has a method (`User.get_role`) specifically for this. You may want to use it instead of another function, which I think is now useless